### DISCUSSION TITLE
Can't delete chat (dev branch)

### DISCUSSION THREAD

**Discussion by bgeneto** [open-webui/open-webui#5860] [General]
# Bug Report

## Installation Method

Docker compose linux

## Environment

- **Open WebUI Version:** v0.3.30
- **Ollama (if applicable):** 0.3.12

- **Operating System:** Debian 12
- **Browser (if applicable):** Chrome 120.0

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:
The user should be able to delete a chat successfully, and the system should handle the deletion request without errors.

## Actual Behavior:
When attempting to delete a user chat, the system raises a KeyError exception due to a missing configuration key. The error occurs in the delete_chat_by_id function in the chats.py file.

## Description

**Bug Summary:**
The delete_chat_by_id function in the chats.py file attempts to access a configuration key that does not exist, resulting in a KeyError exception. This prevents the user from deleting a chat successfully.

## Reproduction Details

**Steps to Reproduce:**

Start the Open WebUI application.
0. Create a new user (non-admin)
1. Log in as the just created user (ensure chat deletion permissions: USER_PERMISSIONS_CHAT_DELETION=true).
2. Navigate to the chat interface and select a chat to delete (...).
3. Click the "Delete" button to initiate the deletion process.
4. The system will raise a KeyError exception in the delete_chat_by_id function, located in the chats.py file, due to the missing configuration key.

## Logs and Screenshots

**Browser Console Logs:**

**Docker Container Logs:**
```text

  File "/usr/local/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 72, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backend/open_webui/apps/webui/routers/chats.py", line 295, in delete_chat_by_id
    if not request.app.state.config.USER_PERMISSIONS["chat"]["deletion"]:
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
KeyError: 'chat'

```

**Screenshots/Screen Recordings (if applicable):**

![image](https://github.com/user-attachments/assets/cf24fc5c-7c2d-4b30-b9c6-8861c88a9849)

## Additional Information
I'm using de dev branch pulled today (2024-09-30)

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by silentoplayz** [comment]
I'm personally unable to replicate the error with the reproduction steps provided. I am also on the dev branch using the latest commit.

**Comment by tjbck** [comment]
Cannot reproduce, but just refactored the code so see if you can get it working with the latest dev!

**Reply by bgeneto** [reply]
> Cannot reproduce, but just refactored the code so see if you can get it working with the latest dev!

Interesting... now I'm getting the notification message below while trying chat deletion with a user (non-admin) account: 

```
You do not have permission to access this resource. Please contact your administrator for assistance.
```

<img width="262" alt="image" src="https://github.com/user-attachments/assets/0ea58ac3-d21e-4eac-aee3-2c8bdf9403d5">

**Reply by silentoplayz** [reply]
> > Cannot reproduce, but just refactored the code so see if you can get it working with the latest dev!
> 
> Interesting... now I'm getting the notification message below while trying chat deletion with a user (non-admin) account:
> 
> ```
> You do not have permission to access this resource. Please contact your administrator for assistance.
> ```
> 
> <img alt="image" width="262" src="https://private-user-images.githubusercontent.com/473074/372653326-0ea58ac3-d21e-4eac-aee3-2c8bdf9403d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4MTc5NjQsIm5iZiI6MTcyNzgxNzY2NCwicGF0aCI6Ii80NzMwNzQvMzcyNjUzMzI2LTBlYTU4YWMzLWQyMWUtNGVhYy1hZWUzLTJjOGJkZjk0MDNkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwMVQyMTIxMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMWQyNjIwZGIzMjM4OGVhNjI3NjczZWI2ZTYzNGM4ZDhkMzljMzM1NjlhYTMzZmM0OGVkNTE3NDVmNDE5NDU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DxR-FrSanaqipo2aBWFJEU9ig434pzBskatkNPUfgYo">

This is very much possible if an Admin account has set `Allow Chat Deletion` to `Don't Allow`. Are you sure this option isn't set to `Don't Allow` on your end from an Admin account?
![image](https://github.com/user-attachments/assets/766de58e-3a3a-413f-b665-b8dafb8f6bc3)

**Reply by bgeneto** [reply]
Well... I think maybe something went wrong for me between updates... 
It seems that I can't even change those settings anymore! 

![image](https://github.com/user-attachments/assets/14d4216f-5ae7-4f21-a472-05adc43229f7)

![image](https://github.com/user-attachments/assets/b2cadc98-e7f2-4c54-b08b-60305c915e11)

And this object:  `request.app.state.config.USER_PERMISSIONS` is empty for every (non-admin) user, that's why it triggers the notification/error message.

Are there any methods available to me to restore my database to a "healthy" condition, in terms of users' permissions at least? TIA.

**Reply by bgeneto** [reply]
`RESET_CONFIG_ON_START=true` did the trick (enabling User Permissions options again). But I had to reconfigure all the (lost) settings manually because when I imported all my json (config, prompts etc...) files it locked those 3 options again.

**Comment by rossm-mf** [comment]
We have the same issue but wanted to see with you @bgeneto about that part

> because when I imported all my json (config, prompts etc...) files it locked those 3 options again.

Do you mean that even after fixing the issue with RESET then import, you got back to the same problem?

**Reply by bgeneto** [reply]
> We have the same issue but wanted to see with you @bgeneto about that part
> 
> > because when I imported all my json (config, prompts etc...) files it locked those 3 options again.
> 
> Do you mean that even after fixing the issue with RESET then import, you got back to the same problem?

Yes. That's right. After reset, when importing my settings it locked again. Maybe a faulty config file on my side, I really don't know what happened... (but it is not the first time).

**Comment by rossm-mf** [comment]
you gave me an idea with your test

I have a staging environment that I compared the saved config with prod

saw that on prod I was missing some values.  I had an empty permission object as this

```
"ui": {
        "default_models": "casperhansen/llama-3-70b-instruct-awq",
        "user_permissions": {
        }
    }
```

I added this

```
"ui": {
        "default_models": "casperhansen/llama-3-70b-instruct-awq",
        "default_user_role": "user",
        "enable_community_sharing": false,
        "enable_message_rating": false,
        "enable_signup": false,
        "user_permissions": {
            "chat": {
                "deletion": true,
                "editing": true,
                "temporary": true
            }
        }
    }
```

that seemed to fix the problem.
